src/citrine/resources/project.py

```diff
@@ -139,9 +141,12 @@ def generative_design_executions(self) -> GenerativeDesignExecutionCollection:
         return GenerativeDesignExecutionCollection(project_id=self.uid, session=self.session)
 
     @property
+    @deprecation.deprecated(deprecated_in="3.2.0", removed_in="4.0.0",
```

Comment:
@kroenlein how does it look? Is the right way of deprecating this?. I gues we'll need to jump to version 4 as removing the dataset collection from the project entity is a breaking change

Reply:
The deprecation syntax is correct, but given that projects are owned by teams, is there a reason to remove this route?  i.e., is there a failure mode where in the new paradigm, a user with a `project` object wouldn't get the right dataset if they said `project.datasets.list()` and that routed to `teams/{team}/datasets`?

Reply:
My preferred approach here would be to add a property for project.team which returns the parent Team of the Project, and then set an alias within the Project class of `project.datasets = project.team.datasets`. This would be both easy to maintain and read.